client.go

```diff
@@ -139,8 +167,63 @@ func (c *Client) Connect(ctx context.Context) error {
 		return err
 	}
 
+	// Start connection monitor in the background
+	// This will help detect silent disconnects and trigger reconnection attempts
+	go c.monitorConnection(2 * time.Second)
+
 	return nil
+}
+
+// monitorConnection periodically checks the gRPC connection state
+// and attempts to reconnect if the connection appears to be dead
+func (c *Client) monitorConnection(checkInterval time.Duration) {
+	// Use default interval if not specified
+	if checkInterval <= 0 {
+		checkInterval = 10 * time.Second
+	}
 
+	ticker := time.NewTicker(checkInterval)
+	defer ticker.Stop()
+
+	for {
+		<-ticker.C
+
+		// Check connection state
+		state := c.conn.GetState()
+		if state != connectivity.Ready && state != connectivity.Connecting && state != connectivity.Idle {
+			// If connection is in a bad state, attempt to reset it
+			c.conn.ResetConnectBackoff()
+
+			// Create a context with timeout for reconnection
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+
+			// Try to force state change
+			if !c.conn.WaitForStateChange(ctx, state) {
+				// If state doesn't change, connection might be dead
+				// We should close it to trigger reconnection in the stream methods
+				c.conn.Close()
+
+				// Attempt to create a new connection
+				newCtx, newCancel := context.WithTimeout(context.Background(), 15*time.Second)
+				newConn, err := grpc.DialContext(
+					newCtx,
+					c.target,
+					grpc.WithTransportCredentials(insecure.NewCredentials()),
+					grpc.WithBlock(),
+				)
+
+				if err == nil {
+					// Successfully created new connection
+					c.conn = newConn
+					c.client = api.NewAPIClient(newConn)
```

Comment:
Will this also reconnect any existing streams that are open? Or should we do that manually?